pkg/utils/authcheck/authtype.go

```diff
@@ -52,12 +55,15 @@ const (
 	BrokerServiceAccountName         = "broker"
 )
 
-var BrokerSecret = &corev1.SecretKeySelector{
-	LocalObjectReference: corev1.LocalObjectReference{
-		Name: "google-broker-key",
-	},
-	Key: "key.json",
-}
+var (
+	// Regex for a valid google service account email.
+	emailRegexp = regexp.MustCompile(`^[a-z][a-z0-9-]{5,29}@[a-z][a-z0-9-]{5,29}\.iam\.gserviceaccount.com$`)
```

Comment:
This isn't quite accurate, on one of my older projects named `example.com:harwayne-example`, I have an GSA with the email `bob123@harwayne-example.example.com.iam.gserviceaccount.com`. I don't think new projects can get this style name, but older ones can. You might want to allow anything between `@{5,29}@[a-z][a-z0-9-]{5,29}` and `.iam.gserviceaccount.com`. Maybe `^[a-z][a-z0-9-]{5,29}@[a-z][a-z0-9-]{5,29}.*\.iam\.gserviceaccount\.com$`.

Reply:
Good to know! Thanks